Added Polylith architecture tool

steffan-westcott opened the pull request:
[Polylith](https://github.com/polyfy/polylith) is a tool used to develop Polylith based architectures in Clojure. See also the [Polylith high level overview](https://polylith.gitbook.io/polylith/)

Please note this is not an application framework, hence the choice of category `Application Architectures`.

Fixes #393

weavejester commented:
What's the difference between an application framework and an "application architecture"?

steffan-westcott commented:
@tengstrand can supply a better answer me, but here's my take. A Polylith architecture is a discipline for structuring program code (a folder structure), but does not provide nor prescribe any runtime library to use. Polylith seems to be addressing concerns distinct to those served by, say, Integrant and Component.

weavejester commented:
I'm reluctant to have a unique category for polylith, particularly if it's a category that needs to be explained by knowing about polylith in the first place.

steffan-westcott commented:
That is fair enough, I have changed the category to `Application Frameworks` as that is the most applicable of the existing ones.

tengstrand commented:
I understand why you don't want to put it in its own category. I think people will find Polylith if we put it under `Application Framework` even though it's not a framework, but a way of structuring software at the architectural level, with some tooling support.
